dotnet/src/SemanticKernel/Planning/Plan.cs

```diff
@@ -279,30 +280,35 @@ public async Task<Plan> InvokeNextStepAsync(SKContext context)
     /// <inheritdoc/>
     public FunctionView Describe()
     {
-        // TODO - Eventually, we should be able to describe a plan and it's expected inputs/outputs
+        // TODO - Eventually, we should be able to describe a plan and its expected inputs/outputs
         return this.Function?.Describe() ?? new();
     }
 
     /// <inheritdoc/>
-    public Task<SKContext> InvokeAsync(string input, SKContext? context = null, CompleteRequestSettings? settings = null, ILogger? log = null,
+    public Task<SKContext> InvokeAsync(
+        string? input = null,
+        CompleteRequestSettings? settings = null,
+        ILogger? log = null,
         CancellationToken cancellationToken = default)
     {
-        context ??= new SKContext(new ContextVariables(), null!, null, log ?? NullLogger.Instance, cancellationToken);
+        this.State.Update(input);
 
-        context.Variables.Update(input);
+        SKContext context = new(
+            this.State,
+            NullMemory.Instance,
+            null,
+            log ?? NullLogger.Instance,
+            cancellationToken);
```

Comment:
The SKContext ctor isn't validating any of these arguments.  Should it be?

Reply:
Updated with defaults for all values